src/reactviews/pages/QueryResult/resultGrid.tsx

```diff
@@ -196,7 +197,11 @@ const ResultGrid = forwardRef<ResultGridHandle, ResultGridProps>((props: ResultG
                                           columnDef,
                                           dataContext,
                                       );
-                                  } else if (isJsonCell(value) && props.resultSetSummary) {
+                                  } else if (
```

Comment:
How about we render all cells as plain text by default, and allow users to double-click (or press Enter/Space) to open a Fluent dialog with a Monaco editor?

1. On double-click or keyboard activation (Enter/Space), open a dialog containing a Monaco editor.
2. Parse the cell’s value to detect its data type.
3. Set the Monaco editor’s language mode based on the detected type.

This approach keeps the grid lightweight and delays the parsing to until user expand the cell. 


<img width="975" height="665" alt="Image" src="https://github.com/user-attachments/assets/9b48de07-c1ec-445f-a187-841ed69570d3" />

However, it is a bigger feature work than this fix.

@kburtram and @croblesm, what do you think?